src/CryptState.h

```diff
@@ -32,23 +28,55 @@ class CryptState {
 		unsigned int uiRemoteLost;
 		unsigned int uiRemoteResync;
 
-		AES_KEY	encrypt_key;
-		AES_KEY decrypt_key;
 		Timer tLastGood;
 		Timer tLastRequest;
 		bool bInit;
 		CryptState();
+		virtual ~CryptState() {};
+
+		virtual bool isValid() const = 0;
+		virtual void genKey() = 0;
+		virtual void setKey(const std::string& rkey, const std::string& eiv, const std::string& div) = 0;
+		virtual void setRawKey(const std::string& rkey) = 0;
+		virtual void setEncryptIV(const std::string& iv) = 0;
+		virtual void setDecryptIV(const std::string& iv) = 0;
+		virtual std::string getRawKey() = 0;
+		virtual std::string getEncryptIV() = 0;
+		virtual std::string getDecryptIV() = 0;
 
-		bool isValid() const;
-		void genKey();
-		void setKey(const unsigned char *rkey, const unsigned char *eiv, const unsigned char *div);
-		void setDecryptIV(const unsigned char *iv);
+		virtual bool decrypt(const unsigned char *source, unsigned char *dst, unsigned int crypted_length) = 0;
+		virtual void encrypt(const unsigned char *source, unsigned char *dst, unsigned int plain_length) = 0;
+};
+
+class CryptStateOCB2 : public CryptState {
```

Comment:
imo this should be a separate file. I know this is not what most other code in Mumble does, but this makes it a lot cleaner.

My suggestion would be to add a sub-directory called e.g. `crypto` and then put all crypto-classes in there (each class in a separate file)

Reply:
Yes. I cannot agree more ;). Can I make a new folder called `crypto` for all crypto-related stuff?

Reply:
Absolutely

Reply:
@davidebeatrici Actually I'm not very sure about creating a new directory. I'm having a hard time modifying all the `qmake` files. Will it be a good idea to leave this kind of works till we have moved to `cmake`?

Reply:
Sounds good.